src/main/webapp/app/exercises/programming/assess/repo-export/programming-assessment-repo-export-dialog.component.ts

```diff
@@ -55,65 +50,44 @@ export class ProgrammingAssessmentRepoExportDialogComponent implements OnInit {
             normalizeCodeStyle: false, // disabled by default because it is rather unstable
             hideStudentNameInZippedFolder: false,
         };
-        this.isRepoExportForMultipleExercises = this.selectedProgrammingExercises.length > 0;
-        if (!this.isRepoExportForMultipleExercises) {
-            this.exerciseService
-                .find(this.exerciseId)
-                .pipe(
-                    tap(({ body: exercise }) => {
-                        this.exercise = exercise!;
-                    }),
-                    catchError((err) => {
-                        this.alertService.error(err);
-                        this.clear();
-                        return of(undefined);
-                    }),
-                )
-                .subscribe(() => {
-                    this.isLoading = false;
-                });
-        } else {
-            this.isLoading = false;
-        }
+        this.isRepoExportForMultipleExercises = this.programmingExercises.length > 1;
+        this.isAtLeastInstructor = this.programmingExercises.every((exercise) => exercise.isAtLeastInstructor);
+        this.isLoading = false;
     }
 
     clear() {
         this.activeModal.dismiss('cancel');
     }
 
-    exportRepos(exerciseId?: number) {
-        if (!exerciseId) {
-            return;
-        }
-        this.exportInProgress = true;
-        // The participation ids take priority over the participant identifiers (student login or team names).
-        if (this.participationIdList?.length) {
-            // We anonymize the assessment process ("double-blind").
-            this.repositoryExportOptions.addParticipantName = false;
-            this.repositoryExportOptions.hideStudentNameInZippedFolder = true;
-            this.repoExportService.exportReposByParticipations(exerciseId, this.participationIdList, this.repositoryExportOptions).subscribe({
+    exportRepos() {
+        this.repositoryExportOptions.exportAllParticipants = true;
+        this.programmingExercises.forEach((exercise) => {
+            if (!exercise.id) {
+                return;
+            }
+            this.exportInProgress = true;
```

Comment:
As far as I can see, you only set this back to false in case of an error. Is that intended?

Reply:
`this.exportInProgress` is also set back to true in the `handleExportRepoResponse` function below... The logic here has basically remained untouched, but only moved within a `forEach` 🙂